packages/react-dom-bindings/src/client/ReactDOMHostConfig.js

```diff
@@ -895,6 +909,20 @@ function getNextHydratable(node) {
             }
             break;
           }
+          case 'STYLE': {
+            const styleEl: HTMLStyleElement = (element: any);
+            if (styleEl.hasAttribute('data-rprec')) {
```

Comment:
Unrelated but I can't help but think that this should just be `data-precedence` to aid debugging. It's not short enough to warrant obfuscation. It might conflict but so could many thing.

Reply:
yeah I think this makes sense. One way to maybe make this takeover less frustrating is to make the opt-in use `data-precedence` rather than `precedence` as a prop. At least it will signal to the writer why we claim that data attribute